selfdrive/camerad/imgproc/utils.cc

```diff
@@ -0,0 +1,76 @@
+#include "utils.h"
+#include <stdio.h>
+
+uint16_t clamp_uint16(float x) {
```

Comment:
I would use a template here.

```c++
template <T>
T clamp(float x) {
  if (x < 0) return 0;
  if (x > std::numeric_limits<T>::max) return std::numeric_limits<T>::max;
  return (T)x;
}

``

Reply:
does this clamp function exist in our cpp version? included algorithm still not found